clang/lib/Sema/SemaOverload.cpp

```diff
@@ -7688,7 +7688,7 @@ bool Sema::CheckNonDependentConversions(
     QualType ParamType = ParamTypes[I + Offset];
     if (!ParamType->isDependentType()) {
       unsigned ConvIdx = PO == OverloadCandidateParamOrder::Reversed
-                             ? 0
+                             ? Args.size() - 1 - (ThisConversions + I)
```

Comment:
Is this a separate bug? Do you have a code example that manifests it independently?
I suggest to move this to a separate commit with its own test if it's not directly connected.

Reply:
Thanks for clearing this up for me offline. After going through an example I finally got what was wrong here.
For reversed operators `Args.size()` is always 2. When `ThisConversions == 0` (i.e. non-methods), we were writing conversions for *both* arguments into `ConvIdx == 0`, after this change we correctly loop through both.
When `ThisConversions == 1`, both versions work the same.

Indeed, the tests you added already check this behavior. Thanks! Resolving this.

Reply:
I think this warrants a comment as to why this computation is correct. It is certainly not obvious and if anyone ever needs to update this code it would be helpful.